agent/consul/leader_metrics.go

```diff
@@ -153,3 +179,14 @@ func initLeaderMetrics() {
 		metrics.SetGaugeWithLabels(g.Name, float32(math.NaN()), g.ConstLabels)
 	}
 }
+
+// expiresSoon checks to see if we are close enough to the cert expiring that
+// we should send out a WARN log message.
+// It returns true if the cert will expire within 28 days or 40% of the
+// certificate's total duration (whichever is shorter).
+func expiresSoon(lifetime, untilAfter time.Duration) bool {
+	defaultPeriod := 28 * (24 * time.Hour) // 28 days
+	fortyPercent := (lifetime / 10) * 4    // 40% of total duration
+
+	return untilAfter < defaultPeriod || untilAfter < fortyPercent
```

Comment:
Is this logically equivalent to the intended behavior, which is `untilAfter < min(defaultPeriod, fortyPercent)`?

Let's say this is for an intermediate CA with a 30 day TTL. Automatic renewal will be first attempted at 50% lifetime (`untilAfter = 15 days`).

Let's say that `untilAfter` is currently 16 days. `fortyPercent` should be 12 days.

`untilAfter < defaultPeriod` -> `16 days < 28 days` -> `true`. The function returns `true`.

The intended behavior is:
`untilAfter < min(defaultPeriod, fortyPercent)` -> `16 days < min(28 days, 12 days)` -> `16 days < 12 days` -> `false`

Reply:
I haven't examined the test cases to check whether it would have caught this. (Maybe there already is such a test case, and there's just something off in my reasoning above.)

Reply:
You are 100% right and this is why I :heart: tests. This made sense on the surface but was slightly different and off... and the tests caught it.